src/pytorch_pretrained_bert/modeling.py

```diff
@@ -1449,10 +1453,15 @@ def forward(self, input_ids, token_type_ids, position_ids, attention_mask, task_
             last_hidden = new_encoded_layers[-1][:, -1:, :]
             prediction_scores, _ = self.cls(
                 last_hidden, None, task_idx=task_idx)
+            prediction_scores = prediction_scores[:, -1, :] / (self.temperature if self.temperature > 0 else 1.)
```

Comment:
`prediction_scores[:, -1, :]` reduces the dimension from 3 to 2, while other places assume the dim=3 (such as `prediction_scores[:, :, token_id].fill_(-10000.0)`).

Reply:
good catch, I checked and prediction_scores only needed to be 3 dimensional for the not_predict_set if conditional, and is not used after that, so I modified the access to work properly with 2 dimensions.

unfortunately due to to way torch.multinomial works I cannot leave it as 3 dimensions, so had to make the modification that way